SessionMessagingKit/SessionUtil/Config Handling/SessionUtil+Contacts.swift

```diff
@@ -0,0 +1,536 @@
+                            [
+                                (data.profile.name.isEmpty || profile.name == data.profile.name ? nil :
+                                    Profile.Columns.name.set(to: data.profile.name)
+                                ),
+                                (profile.nickname == data.profile.nickname ? nil :
+                                    Profile.Columns.nickname.set(to: data.profile.nickname)
+                                ),
+                                (profile.profilePictureUrl != data.profile.profilePictureUrl ? nil :
+                                    Profile.Columns.profilePictureUrl.set(to: data.profile.profilePictureUrl)
+                                ),
+                                (profile.profileEncryptionKey != data.profile.profileEncryptionKey ? nil :
+                                    Profile.Columns.profileEncryptionKey.set(to: data.profile.profileEncryptionKey)
+                                )
+                            ].compactMap { $0 }
+                        )
+                }
+                
+                /// Since message requests have no reverse, we should only handle setting `isApproved`
+                /// and `didApproveMe` to `true`. This may prevent some weird edge cases where a config message
+                /// swapping `isApproved` and `didApproveMe` to `false`
+                if
+                    (contact.isApproved != data.contact.isApproved) ||
+                    (contact.isBlocked != data.contact.isBlocked) ||
+                    (contact.didApproveMe != data.contact.didApproveMe)
+                {
+                    try contact.save(db)
+                    try Contact
+                        .filter(id: sessionId)
+                        .updateAll( // Handling a config update so don't use `updateAllAndConfig`
+                            db,
+                            [
+                                (!data.contact.isApproved || contact.isApproved == data.contact.isApproved ? nil :
+                                    Contact.Columns.isApproved.set(to: true)
+                                ),
+                                (contact.isBlocked == data.contact.isBlocked ? nil :
+                                    Contact.Columns.isBlocked.set(to: data.contact.isBlocked)
+                                ),
+                                (!data.contact.didApproveMe || contact.didApproveMe == data.contact.didApproveMe ? nil :
+                                    Contact.Columns.didApproveMe.set(to: true)
+                                )
+                            ].compactMap { $0 }
+                        )
+                }
+                
+                /// If the contact's `hidden` flag doesn't match the visibility of their conversation then create/delete the
+                /// associated contact conversation accordingly
+                let threadInfo: PriorityVisibilityInfo? = try? SessionThread
```

Comment:
There should be a `.filter(id: sessionId)` I reckon.

Reply:
Yep good catch, have updated

Reply:
Surprised this hasn't caused more bugs actually 🤔

Reply:
Yeah, it's strange that this didn't cause more bugs.